### PR TITLE
Scroll box ranges

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 1]
```diff
@@ -1243,6 +1245,24 @@ UiProfile Spire::make_scroll_box_profile() {
       vertical_display_policy.connect_changed_signal([scroll_box] (auto value) {
         scroll_box->set_vertical(value);
       });
+      auto& horizontal_padding = get<int>("horizontal-padding",
+        profile.get_properties());
+      horizontal_padding.connect_changed_signal([=] (auto padding) {
+        auto style = get_style(*scroll_box);
+        style.get(Any()).
+          set(PaddingLeft(padding)).
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:1253]: ```horizontal_padding```

File: Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp [hunk 2]
```diff
@@ -1243,6 +1245,24 @@ UiProfile Spire::make_scroll_box_profile() {
...
+          set(PaddingRight(padding));
+        set_style(*scroll_box, std::move(style));
+      });
+      auto& vertical_padding = get<int>("vertical-padding",
+        profile.get_properties());
+      vertical_padding.connect_changed_signal([=] (auto padding) {
+        auto style = get_style(*scroll_box);
+        style.get(Any()).
+          set(PaddingTop(padding)).
```
Comment (kranar) [inline Applications/Spire/Source/UiViewer/StandardUiProfiles.cpp:1262]: ```vertical_padding```

**Review by kranar — CHANGES_REQUESTED** [review]

**Review by shaobinspire — COMMENTED** [review]
When the padding changes, do the ranges need to be updated?

**Review by kranar — APPROVED** [review]
This is fantastic, I am left speechless. Very straight forward solution to a difficult problem and the demo is excellent.

**Pull request merged**